dataUtil/featureExtraction.py

```diff
@@ -0,0 +1,114 @@
+import numpy as np
+from scipy.signal import spectrogram as spectro
+from dataUtil.processing import read_audio_data
+from librosa.feature import mfcc
+from librosa.display import specshow
+import matplotlib.pyplot as plt
+
+
+def get_fft(signal, sampling_frequency):
+    '''
+    This function takes in an array, either of size (num_frames,) or (num_batches, num_frames)
+        and returns a tuple of times, frequencies, and a spectrogram.
+    :param signal: the signal to get the spectrogram from
+    :param sampling_frequency: the sampling frequency of the signal
+    :param window_size: the number of samples per time segment (column) of the spectrogram
+    :return: a tuple of segment times (shape: [num_batches,] num_segments),
+                        sample frequncies (shape: [num_baches,] num_frequencies),
+                        spectrogram (shape: [num_batches,] num_segments, num_frequencies)
+    '''
+    frequencies, times, spectrogram = spectro(signal, fs=sampling_frequency, nfft=2048)
+    # Only return important frequencies
+    return times, frequencies[:70], np.swapaxes(spectrogram, 1, 2)[:, :, :70]
+
+
+def _get_single_mfcc(signal, sampling_frequency, num_mfcc):
+    '''
+    This function takes in an array of size (num_frames,) and returns the mfccs of size (num_segments, num_mfcc)
+    :param signal: the signal to get the mfccs from
+    :param sampling_frequency: the sampling frequency of the signal
+    :param num_mfcc: the number of mfccs to extract
+    :return: an array of size (num_segments, num_mfcc)
+    '''
+    return mfcc(y=signal, sr=sampling_frequency, n_mfcc=num_mfcc).T
+
+def get_mfcc(signals, sampling_frequency, num_fcc):
+    '''
+    This function takes in a batch of signals of size (num_batches, num_frames,)
+        and returns the mfccs of size (num_batches, num_segments, num_mfcc)
+    :param signals: the signal to get the mfccs from
+    :param sampling_frequency: the sampling frequency of the signal
+    :param num_fcc: the number of mfccs to extract
+    :return: an array of size (num_batches, num_segments, num_mfcc)
+    '''
+    mfccs = []
+    for signal in signals:
+        mfccs.append(_get_single_mfcc(signal, sampling_frequency, num_fcc))
```

Comment:
Is there a case when `_get_single_mfcc` can return an empty array? If so, I believe `np.array(mfccs)` will fail.